src/main/java/uk/gov/justice/digital/job/DataHubJob.java

```diff
@@ -111,41 +121,46 @@ public void batchProcessor(JavaRDD<byte[]> batch) {
                     val dataFrameForTable = extractDataFrameForSourceTable(dataFrame, tableInfo);
                     dataFrameForTable.persist();
 
-                    rawZone.process(spark, dataFrameForTable, tableInfo);
+                    String sourceName = tableInfo.getAs(SOURCE);
+                    String tableName = tableInfo.getAs(TABLE);
 
-                    val structuredLoadDataFrame = structuredZoneLoad.process(spark, dataFrameForTable, tableInfo);
-                    val structuredIncrementalDataFrame = structuredZoneCDC.process(spark, dataFrameForTable, tableInfo);
+                    val optionalSourceReference = sourceReferenceService.getSourceReference(sourceName, tableName);
```

Comment:
The source reference is retrieved only once and applied to the structured zone, curated zone and domain refresh.